File: guides/micronaut-mongodb-synchronous/java/src/main/java/example/micronaut/FruitRepository.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import java.util.List;

public interface FruitRepository {

    @NonNull
    List<Fruit> list();

    void save(@NonNull @NotNull @Valid Fruit fruit); // <1>
}

File: guides/micronaut-mongodb-synchronous/python/src/example/micronaut/fruit_repository.py
from abc import ABC, abstractmethod
from typing import Annotated

from jakarta.validation import Valid

from .fruit import Fruit


class FruitRepository(ABC):

    @abstractmethod
    def list(self) -> list[Fruit]:
        pass

    @abstractmethod
    def save(self, fruit: Annotated[Fruit, Valid]) -> None:  # <1>
        pass
